content/posts/replacing-the-redisgraph-parser.md

```diff
@@ -20,15 +20,15 @@ Since its inception, RedisGraph used [Flex](https://github.com/westes/flex) and
 
 > The extra effort that it takes to add new features is the interest paid on the debt.
 
-To use [Martin Fowler's language of technical debt](https://martinfowler.com/bliki/TechnicalDebt.html), our self-maintained grammar charged us interest every time we introduced or modified a syntactical construct. Gradual improvement was not an option, however, so we were reluctant to undertake the effort of a full replacement. We finally relented when working on the WITH clause, which is how Cypher allows for multiple queries to be chained into a single atomic execution. Unlike subqueries in SQL, which create a tree of independent executions, the WITH clause flattens query parts into a linear composition:
+To use [Martin Fowler's language of technical debt](https://martinfowler.com/bliki/TechnicalDebt.html), our self-maintained grammar charged us interest every time we introduced or modified a syntactical construct. Gradual improvement was not an option, however, so we were reluctant to undertake the effort of a full replacement. We finally relented when working on the `WITH`clause, which is how Cypher allows for multiple queries to be chained into a single atomic execution. Unlike subqueries in SQL, which create a tree of independent executions, the `WITH` clause flattens query parts into a linear composition:
 
-![Multi-Part Query Railroad Diagram](/multipart_query.png)
+![Multi-Part Query Railroad Diagram](/MultiPartQuery.svg)
 [Cypher railroad diagram for multi-part queries](https://s3.amazonaws.com/artifacts.opencypher.org/railroad/MultiPartQuery.html)
 
 Our original grammar was effective at interpreting a sequence of clauses as a self-contained query, but it would have required a major rewrite to interpret that same sequence as one scope capable of reading from and projecting into others. libcypher-parser had already solved this problem and many more, so we started the even more ambitious rewrite of scrapping our parser and building off an open source project.
 
 ## The joy of FOSS
-libcypher-parser was a perfect fit for our needs. Beyond the core criteria of being a feature-rich Cypher parser, it is a C implementation, stable but evolving, and—above all—open source.
+libcypher-parser was a perfect fit for our needs. Beyond the core criteria of being a feature-rich Cypher parser, it is a C implementation, stable but evolving, and — above all — open source.
```

Comment:
I couldn't tell before whether we are supposed to use spaces around em-dashes - @haleykim100, notes on this?

Reply:
According to our Style Guide, we do not use spaces around em-dashes!